source/2019-12-15-countdown-to-the-new-year-2019-ember-test-selectors.md

```diff
@@ -0,0 +1,25 @@
+---
+title: Countdown to The New Year: Ember Test Selectors
+author: Jessica Jordan
+tags: 2019, Addons, DecEmber, Testing
+alias: 2019/12/15/countdown-to-the-new-year-2019-ember-test-selectors
+responsive: true
+---
+
+Our December series - "Countdown to The New Year: 31 Days of Ember Addons" aims to highlight a new addon each day until the new year, and we hope you enjoy hearing about addons that bring us joy as well!
+
+## Day 15
+
+On the 15th day of our **Countdown to The New Year** we want to take a look at a helpful Ember addon which lets us write **redesign-proof tests**: [Ember Test Selectors](https://emberobserver.com/addons/ember-test-selectors)!
+
+### What It Does
+
+Ember Test Selectors is an addon that allows us to apply additional markup to your templates and components as `data-test` attributes. This additional, test-specific markup allows us to select elements in our integration and application tests reliably - even if the HTML markup or CSS selectors change during a redesign of our app.
```

Comment:
I don't know how much you want to go into detail here, but you can generally use `data-test` attributes even without the addon. The main thing that the addon does is stripping those attributes out of the production builds so that our users don't have to pay for them when they download the JS files.

Reply:
Yeah you're right, I didn't focus so much on that and in an Ember Octane world the automatic attribute binding on components is also not so relevant anymore? I can shift the focus of this section a bit

Reply:
I reworded the section a bit to make the main benefit of the addon clearer that you mentioned in your previous comment - lmk what you think!